test/unit/math/prim/mat/prob/vector_rng_test_helper.hpp

```diff
@@ -0,0 +1,364 @@
+#include <stan/math/prim/mat.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <test/unit/math/prim/scal/prob/util.hpp>
+
+enum class Constraint { None, Positive, NonNegative };
```

Comment:
I think these constraints cover most of the rngs that work on (-inf, inf) and (0, inf).